### PR TITLE
[feature] add Review functionalities

### PR DESCRIPTION
**Pull request by astrohsy**
- [x] ReviewController
- [x] ReviewService
- [x] ReviewControllerTest
- [x] ReviewServiceTest

### CODE REVIEW

**Comment by astrohsy** [comment]
Special operations:

- [ ] post review
- [ ] search based on title of location entity
- [ ] provide sort by ratings

File: api_server/src/main/java/io/zetch/app/domain/review/ReviewDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import io.zetch.app.domain.location.LocationDto;
+import io.zetch.app.domain.user.UserDto;
+import java.io.Serializable;
+import lombok.*;
+
+@Value
+@Builder
+@AllArgsConstructor
+public class ReviewDto implements Serializable {
```
Comment (zagbapu) [inline api_server/src/main/java/io/zetch/app/domain/review/ReviewDto.java:11]: I believe all the dtos have the id field removed.

File: api_server/src/main/java/io/zetch/app/domain/review/ReviewPostDto.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import java.io.Serializable;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.ToString;
+import lombok.Value;
+
+@Value
+@Builder
+@AllArgsConstructor
+public class ReviewPostDto implements Serializable {
```
Comment (zagbapu) [inline api_server/src/main/java/io/zetch/app/domain/review/ReviewPostDto.java:12]: Is this different than ReviewDto

**Review by zagbapu — APPROVED** [review]

File: api_server/src/main/java/io/zetch/app/domain/review/ReviewPostDto.java (hunk 2 above)
Comment (astrohsy) [reply api_server/src/main/java/io/zetch/app/domain/review/ReviewPostDto.java:12]: I thought about passing only `location_id`, `user_id` in POST `/reviews`, but fetching concrete `location`, `user` for the GET `/reviews`.

I now see that we can make it more concise if we define the base dto, `ReviewDto` that includes `comment` and `rating`, and then make `ReviewPostDto` and `ReviewGetDto` inherited by that.

What do you think about this idea?

File: api_server/src/main/java/io/zetch/app/domain/review/ReviewDto.java (hunk 1 above)
Comment (astrohsy) [reply api_server/src/main/java/io/zetch/app/domain/review/ReviewDto.java:11]: I think every Dtos except `Post` should have ids. Without knowing Ids using APIs, we cannot create joined instances.

**Pull request merged**